src/main/c/seasocks/Connection.h

```diff
@@ -70,6 +70,8 @@ class Connection : public WebSocket {
     virtual void send(const char* webSocketResponse) override;
     virtual void send(const uint8_t* webSocketResponse, size_t length) override;
     virtual void close() override;
+    virtual void setUserdata(std::shared_ptr<void> udata) override { _userdata = udata; }
+    virtual std::shared_ptr<void> getUserdata() const override { return _userdata; }
 
     // From Request.
     virtual std::shared_ptr<Credentials> credentials() const override;
```

Comment:
So....in all the cases I use the credentials to hang arbitrary dat (key/value for authentication tokens, groups etc). If you need something with actual RAII-style/object semantics then have user data does make sense.